EKF/ekf_helper.cpp

```diff
@@ -1775,8 +1775,14 @@ bool Ekf::resetYawToEKFGSF()
 
 		// record a magnetic field alignment event to prevent possibility of the EKF trying to reset the yaw to the mag later in flight
 		_flt_mag_align_start_time = _imu_sample_delayed.time_us;
+		_control_status.flags.mag_aligned_in_flight = true;
```

Comment:
This isn't true.
The definition of `mag_aligned_in_flight` is "true when the in-flight mag field alignment has been completed".
I understand that this variable needs to be set to true to have it working, but we then need to rename that variable and make sure that is doesn't have other side effects. For example [getMagDeclination](https://github.com/PX4/ecl/blob/d09473da89ca7f56fcbfe5d3eea4c3e1ca84fa25/EKF/ekf_helper.cpp#L601-L621) assumes that `mag_aligned_in_flight` means that the earth mag field states of the EKF can now be used to compute the declination.
Misusing that variable will create corner cases that will be extremely difficult to debug